processor/cloningfanoutconnector.go

```diff
@@ -150,7 +233,16 @@ func cloneTraceData(td *consumerdata.TraceData) *consumerdata.TraceData {
 	return clone
 }
 
-func cloneMetricsData(md *consumerdata.MetricsData) *consumerdata.MetricsData {
+func cloneTraceData(td data.TraceData) data.TraceData {
```

Comment:
I would add this "Clone" to the MetricData, just in case we will have more than the ResourceSpans list. Can be done in a separate PR.

Reply:
moved